packages/app/src/sandbox/compile.ts

```diff
@@ -794,6 +794,13 @@ async function compile(opts: CompileOptions) {
         }
       }
     }, 600);
+
+    /**
+     * For tracking purpose
+     */
+    document.addEventListener('click', () => {
+      dispatch({ type: 'document-focus' });
+    });
```

Comment:
Does it make sense to place this piece of code here?

Reply:
Not really, it's pretty confusing and this function needs a ton of cleanup. But as far as I know it gets called on every compilation so even on hot reloads, resulting in a ton of event listeners at some point.

This is also true for the `initializeDOMMutationListener` function call at the bottom.

The correct place for this and the other function I mentioned would be `packages/app/src/sandbox/index.js`